### ISSUE TITLE
Setting ConfigDockingTabBarOnSingleWindows prevents windows from auto-resizing on first display

### ISSUE THREAD

**Issue by haldean** [ocornut/imgui#2386]
**Version/Branch of Dear ImGui:**

Version: 1.69 WIP
Branch: docking (commit e7c8dd37b3ddaa98e651bec3a1679e57b356719c)

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Compiler: g++ 8.2.1
Operating System: Arch Linux

**My Issue/Question:**

Setting `io.ConfigDockingTabBarOnSingleWindows` to true causes windows to appear in their smallest possible size, sometimes after a single frame of them being correctly sized. After the initial display, you can resize the windows using the window handles as expected.

**Screenshots/Video**

Both of these screenshots are taken immediately after the application starts, with no user interaction required.

With `io.ConfigDockingTabBarOnSingleWindows = false`:

![2019-02-26-123930_1264x744_scrot](https://user-images.githubusercontent.com/187501/53444709-b9325400-39c3-11e9-98a9-2be05fb69a3d.png)

With `io.ConfigDockingTabBarOnSingleWindows = true`:

![2019-02-26-123917_1180x758_scrot](https://user-images.githubusercontent.com/187501/53444708-b9325400-39c3-11e9-8671-4484a2e9b672.png)

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
int main(void) {
  if (!glfwInit()) {
    std::cerr << "[E] init glfw" << std::endl;
    return EXIT_FAILURE;
  }

  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
  glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
  glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);

  GLFWwindow *window = glfwCreateWindow(800, 800, "ana", NULL, NULL);
  if (!window) {
    std::cerr << "[E] failed to create window" << std::endl;
    return EXIT_FAILURE;
  }

  glfwMakeContextCurrent(window);
  GLenum glew_status = glewInit();
  if (glew_status != GLEW_OK) {
    std::cerr << "[E] glewInit: " << glewGetErrorString(glew_status) << std::endl;
    return EXIT_FAILURE;
  }

  ImGui::CreateContext();
  ImGuiIO &io = ImGui::GetIO();

  // to make sure the window sizes aren't coming from the INI file
  io.IniFilename = nullptr;
  io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;
  io.ConfigDockingTabBarOnSingleWindows = true; // if this line is removed the bug goes away

  ImGui_ImplGlfw_InitForOpenGL(window, true);
  ImGui_ImplOpenGL3_Init("#version 410");

  glClearColor(0.0f, 0.0f, 0.0f, 1.0f);

  while (!glfwWindowShouldClose(window)) {
    glfwPollEvents();

    ImGui_ImplOpenGL3_NewFrame();
    ImGui_ImplGlfw_NewFrame();
    ImGui::NewFrame();

    ImGui::ShowDemoWindow();

    ImGui::Begin("Hello world");
    ImGui::Text("Hello!");
    ImGui::End();

    ImGui::Render();

    glClear(GL_COLOR_BUFFER_BIT);
    ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
    glfwMakeContextCurrent(window);
    glfwSwapBuffers(window);
  }

  ImGui_ImplOpenGL3_Shutdown();
  ImGui_ImplGlfw_Shutdown();
  glfwTerminate();
  return EXIT_SUCCESS;
}

```

**Comment by ocornut** [comment]
Thanks. This a known issue with docking nodes.
The `io.ConfigDockingTabBarOnSingleWindows` only makes this more visible.

Other issues are:
- [ ] Default size of first frame is incorrect
- [ ] Auto-resizing is not honored.
- [ ] Cannot collapse.
- [ ] Resize grip is under the scrollbar.
- [ ] Unable to navigate from menu bar to tabs.

I've been working on this and have unfinished patches for some of them but they are not trivial changes.

**Comment by ocornut** [comment]
I have pushed a fix for one of the issue highlighted above in 9e294be (initial size of the docking node) which was arguably the most annoying of the lots.

I'm not satisfied with the patch because it always start with the size of the first window, which is the most common case with `io.ConfigDockingAlwaysTabBar` but it should start with the size of the _focused_ window in case two are created with the same docking node. It's not too bad because that would only happen if there was no saved settings.

Unfortunately fixing this led me to refactor a bunch of technical debt with handling data flow between hosting window, hosted windows, and dock node structure and I don't have a satisfying patch for the full thing yet (and I've sunk lots of time on it already :( ).

TL;DR; the specific case reported above is fixed for the most common case. 
I'll keep the issue open because eventually I want to make single-window dock-node have zero regression compared to regular floating windows.

PS: please note that `io.ConfigDockingTabBarOnSingleWindows` was renamed to `io.ConfigDockingAlwaysTabBar` about a month ago.
